tests/functional/user-data-manager.test.js

```diff
@@ -787,4 +788,20 @@ describe('user-data-manager.test.js', () => {
     resultQuery = await localAdapter.query({ languageCode: 'lat' })
     expect(resultQuery.length).toEqual(1)
   }, 50000)
+
+  it('21 UserDataManager - clear unsubscribes events', async () => {
+    let mockUnsub = jest.fn()
+    let mockEvent = {
+      sub: () => {
+        return mockUnsub
+      }
+    }
+    let udm = new UserDataManager(auth,{
+      WORDITEM_UPDATED: mockEvent,
+      WORDITEM_DELETED: mockEvent,
+      WORDLIST_DELETED: mockEvent})
+    expect(udm.subscriptions.length).toEqual(3)
+    udm.clear()
+    expect(mockUnsub).toHaveBeenCalledTimes(3)
```

Comment:
This test verifies that an unsubscribe function is called three times. For this, we have to emulate the internals of the `sub()` method. This method is not a part of the user data manager. I think that probably it's going too deep into internal implementation of a `PSEvent` for the current test's scope. 

Maybe we should not test how it is unsubsribed (i.e. whether it calls `unsub()` method, which on itself, may not guarantee unsubscription if there is an error in there or if it will change in the future) but rather test the results of unsubsription (i.e. that all subsriptions has been removed, that is `expect(udm.subscriptions.length).toEqual(0)`)? `subsriptions` is part of the public API of `PSEvent` so this way we won't touch its innards.

What do you think?

Reply:
yes, you are right. that is a good suggestion. Will make that change. thank you!